DetectorDescription/DDCMS/src/DDFilteredView.cc

```diff
@@ -194,24 +218,115 @@ bool DDFilteredView::firstChild() {
 }
 
+int DDFilteredView::nodeCopyNo(const std::string_view copyNo) const {
+  int result;
+  if (auto [p, ec] = std::from_chars(copyNo.data(), copyNo.data() + copyNo.size(), result); ec == std::errc()) {
+    return result;
+  }
+  return -1;
+}
+
+std::vector<std::pair<std::string_view, int>> DDFilteredView::toNodeNames(const std::string& path) {
+  std::vector<std::pair<std::string_view, int>> result;
+  std::vector<string_view> names = split(path, "/");
+  for (const auto& i : names) {
+    auto name = noNamespace(i);
+    int copyNo = -1;
+    auto lpos = name.find_first_of('[');
+    if (lpos != std::string::npos) {
+      auto rpos = name.find_last_of(']');
+      if (rpos != std::string::npos) {
+        copyNo = nodeCopyNo(name.substr(lpos + 1, rpos - 1));
+      }
+      result.emplace_back(name.substr(0, lpos), copyNo);
+    } else {
+      result.emplace_back(name, -1);
+    }
+  }
+
+  return result;
+}
+
+bool DDFilteredView::match(const std::string& path, const std::vector<std::pair<std::string_view, int>>& names) const {
+  std::vector<std::pair<std::string_view, int>> toks;
+  std::vector<string_view> pnames = split(path, "/");
+  for (const auto& i : pnames) {
+    auto name = noNamespace(i);
+    auto lpos = name.find_first_of('_');
+    if (lpos != std::string::npos) {
+      int copyNo = nodeCopyNo(name.substr(lpos + 1));
+      toks.emplace_back(name.substr(0, lpos), copyNo);
+    }
+  }
+  if (toks.size() != names.size()) {
+    return false;
+  }
+
+  for (unsigned int i = 0; i < names.size(); i++) {
+    if (names[i].first != toks[i].first) {
+      return false;
+    } else {
+      if (names[i].second != -1 and names[i].second != toks[i].second) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+std::vector<std::vector<Node*>> DDFilteredView::children(const std::string& selectPath) {
+  std::vector<std::vector<Node*>> paths;
+  if (it_.empty()) {
+    LogVerbatim("DDFilteredView") << "Iterator vector has zero size.";
+    return paths;
+  }
+  if (node_ == nullptr) {
+    throw cms::Exception("DDFilteredView") << "Can't get children of a null node. Please, call firstChild().";
+  }
+  it_.back().SetType(0);
+  std::vector<std::pair<std::string_view, int>> names = toNodeNames(selectPath);
+  auto rit = names.rbegin();
+  Node* node = it_.back().Next();
+  while (node != nullptr) {
+    if (node->GetVolume()->GetName() == std::string({rit->first.data(), rit->first.size()})) {
+      std::string pathToNode = path();
+      std::string::size_type n = pathToNode.find(node_->GetName());
+      std::string pathFromParent = pathToNode.substr(n);
+
+      if (match(pathFromParent, names)) {
+        std::vector<Node*> children;
+        LogVerbatim("Geometry") << "Match found: " << pathFromParent;
+        for (int i = startLevel_; i < it_.back().GetLevel(); i++) {
+          children.emplace_back(it_.back().GetNode(i));
+        }
+        children.emplace_back(node);
+        paths.emplace_back(children);
```

Comment:
```suggestion
        childNodes.emplace_back(node);
        paths.emplace_back(childNodes);
```
(Trying to suggest two lines being changed.)